errors.go

```diff
@@ -60,6 +60,21 @@ func (e *OperationError) Error() string {
 	return fmt.Sprintf("operation error %s: %s, %v", e.ServiceName, e.OperationName, e.Err)
 }
 
+// DeserializationError provides a wrapper for and error that occurs during
+// deserialization.
+type DeserializationError struct {
+	Err      error //  original error
+	Snapshot []byte
+}
+
+// Error returns a formatted error for DeserializationError
+func (e *DeserializationError) Error() string {
+	return fmt.Sprintf("deserialization failed, %v", e.Err)
+}
```

Comment:
Nit: I have the change in my improvement PR, basically this should check if e.Err is nil. Similar to how SerializationError is doing. 

```
// Error returns a formatted error for DeserializationError
func (e *DeserializationError) Error() string {
	const msg = "deserialization failed"
	if e.Err == nil {
		return msg
	}
	return fmt.Sprintf("%s: %v", msg, e.Err)
}

```

Reply:
cool will update.